### PR TITLE
Issue #14019: Kill mutation for setSeverity in Checker

### PR DESCRIPTION
**Pull request by Zopsss**
Issue #14019 Kill mutation for setSeverity in checker

created a test case to kill the mutation

### CODE REVIEW

File: src/test/java/com/puppycrawl/tools/checkstyle/CheckerTest.java
Comment (romani) [inline src/test/java/com/puppycrawl/tools/checkstyle/CheckerTest.java:1664]: Please keep it, we are still violating rule to not use deprecated method, and it shows us that we can not avoid usage of it all, until we support Checker in inlined config.

Please put this annotation on method that you created.

**Review by romani — CHANGES_REQUESTED** [review]
Very good, minor changes:

**Comment by Zopsss** [comment]
> ⚠️ Failed to start an instance: FAILED_PRECONDITION: Monthly compute limit exceeded!


Cirrus is failing

**Comment by Zopsss** [comment]
> ⚠️ Failed to start an instance: FAILED_PRECONDITION: Monthly compute limit exceeded!
Cirrus is failing

File: src/test/java/com/puppycrawl/tools/checkstyle/CheckerTest.java
Comment (Zopsss) [reply src/test/java/com/puppycrawl/tools/checkstyle/CheckerTest.java]: Done

File: src/test/java/com/puppycrawl/tools/checkstyle/CheckerTest.java
Comment (romani) [inline src/test/java/com/puppycrawl/tools/checkstyle/CheckerTest.java:443]: Please add `@SuppressForbidden`
In separate PRs we do huge effort to avoid usage of `verify`
But this test has no choice  , and ok to use this method

File: src/test/java/com/puppycrawl/tools/checkstyle/CheckerTest.java
Comment (romani) [inline src/test/java/com/puppycrawl/tools/checkstyle/CheckerTest.java:449]: Please add empty line before this line.
It is better to separate configuration/preparation and execution

**Review by romani — CHANGES_REQUESTED** [review]
Items

File: src/test/java/com/puppycrawl/tools/checkstyle/CheckerTest.java
Comment (Zopsss) [reply src/test/java/com/puppycrawl/tools/checkstyle/CheckerTest.java:443]: if I add this, the test file will exceed the limit of 2000 lines, it will start giving error. What should I do to solve this error?

File: src/test/java/com/puppycrawl/tools/checkstyle/CheckerTest.java
Comment (Zopsss) [reply src/test/java/com/puppycrawl/tools/checkstyle/CheckerTest.java:449]: I removed the empty lines in this test cuz the file was exceeding the limit of 2000 lines, it was giving error about it.

File: src/test/java/com/puppycrawl/tools/checkstyle/CheckerTest.java
Comment (Zopsss) [reply src/test/java/com/puppycrawl/tools/checkstyle/CheckerTest.java:443]: error log:
```
[ERROR] [checkstyle] [ERROR] C:\Open Source\checkstyle\src\test\java\com\puppycrawl\tools\checkstyle\CheckerTest.java:1: File length is 2,002 lines (max allowed is 2,000). [FileLength]
```

Comment (romani) [reply src/test/java/com/puppycrawl/tools/checkstyle/CheckerTest.java:443]: Please add it to suppression, it might be not first for suppression .

Search keyword `FileLength` in config folder

Comment (Zopsss) [reply src/test/java/com/puppycrawl/tools/checkstyle/CheckerTest.java:444]: Done

File: src/test/java/com/puppycrawl/tools/checkstyle/CheckerTest.java
Comment (romani) [inline src/test/java/com/puppycrawl/tools/checkstyle/CheckerTest.java:442]: please add comment about a reason why we have `SuppressForbidden` , verify is one of many methods we forbid, so reader should know what was out target.

**Review by romani — CHANGES_REQUESTED** [review]
last minor:

File: src/test/java/com/puppycrawl/tools/checkstyle/CheckerTest.java
Comment (Zopsss) [reply src/test/java/com/puppycrawl/tools/checkstyle/CheckerTest.java:442]: Sorry but can you tell me the reason why we should avoid using verify() method? I don't know what to write in the comment

Comment (romani) [reply src/test/java/com/puppycrawl/tools/checkstyle/CheckerTest.java:442]: Reason is described at https://github.com/checkstyle/checkstyle/issues/11446

Make a comment like:
```
/*
SuppressForbidden is used to allow usage of 'verify' method. We should use it until we add support for Checker properties in inlined config 
*/
```

Comment (Zopsss) [reply src/test/java/com/puppycrawl/tools/checkstyle/CheckerTest.java:442]: okay and I forgot to mention that there are other test methods too in CheckerTest which uses `verify()` but does not have  `@SuppressForbidden` annotation and also don't have the comment you asked me to add... should I add the annotation and comment to those methods too?

Comment (romani) [reply src/test/java/com/puppycrawl/tools/checkstyle/CheckerTest.java:442]: To unblock this PR sooner please do only update for test you updating.

We can create separate PR to update all others to  be in better form.

Comment (romani) [reply src/test/java/com/puppycrawl/tools/checkstyle/CheckerTest.java:442]: We can make to use good verify method if we define config by xml , example https://github.com/checkstyle/checkstyle/commit/3e1be5063f621252da86a09316fe31a7209ea366#diff-00c2dc5d7847e5fac63bbc22c7910ed1d043fc47999674cb2f11c53db985d514R2

In this case we do not need any suppression at all, and life becomes logical and simple. Please try this.

Comment (Zopsss) [reply src/test/java/com/puppycrawl/tools/checkstyle/CheckerTest.java:442]: @romani I tried using `verifyWithInlineConfigParser()` in place of `verify()` as showed in the commit, but the test started failing idk why. Here is the test case:

```java
@Test
public void testSetSeverity() throws Exception {
    final String[] expected = CommonUtil.EMPTY_STRING_ARRAY;

    verifyWithInlineConfigParser(getPath("InputCheckerTestSeverity.java"), expected);
} 
```

**Error log:**

```
com.puppycrawl.tools.checkstyle.api.CheckstyleException: Config comment not specified properly in C:\Open Source\checkstyle\src\test\resources\com\puppycrawl\tools\checkstyle\checker\InputCheckerTestSeverity.java

	at com.puppycrawl.tools.checkstyle.bdd.InlineConfigParser.parse(InlineConfigParser.java:210)
	at com.puppycrawl.tools.checkstyle.bdd.InlineConfigParser.parse(InlineConfigParser.java:190)
	at com.puppycrawl.tools.checkstyle.AbstractModuleTestSupport.verifyWithInlineConfigParser(AbstractModuleTestSupport.java:235)
	at com.puppycrawl.tools.checkstyle.CheckerTest.testSetSeverity(CheckerTest.java:446)
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:104)
	at java.base/java.lang.reflect.Method.invoke(Method.java:578)
	at org.junit.platform.commons.util.ReflectionUtils.invokeMethod(ReflectionUtils.java:728)
	at org.junit.jupiter.engine.execution.MethodInvocation.proceed(MethodInvocation.java:60)
	at org.junit.jupiter.engine.execution.InvocationInterceptorChain$ValidatingInvocation.proceed(InvocationInterceptorChain.java:131)
	at org.junit.jupiter.engine.extension.TimeoutExtension.intercept(TimeoutExtension.java:156)
	at org.junit.jupiter.engine.extension.TimeoutExtension.interceptTestableMethod(TimeoutExtension.java:147)
	at org.junit.jupiter.engine.extension.TimeoutExtension.interceptTestMethod(TimeoutExtension.java:86)
	at org.junit.jupiter.engine.execution.InterceptingExecutableInvoker$ReflectiveInterceptorCall.lambda$ofVoidMethod$0(InterceptingExecutableInvoker.java:103)
	at org.junit.jupiter.engine.execution.InterceptingExecutableInvoker.lambda$invoke$0(InterceptingExecutableInvoker.java:93)
	at org.junit.jupiter.engine.execution.InvocationInterceptorChain$InterceptedInvocation.proceed(InvocationInterceptorChain.java:106)
	at org.junit.jupiter.engine.execution.InvocationInterceptorChain.proceed(InvocationInterceptorChain.java:64)
	at org.junit.jupiter.engine.execution.InvocationInterceptorChain.chainAndInvoke(InvocationInterceptorChain.java:45)
	at org.junit.jupiter.engine.execution.InvocationInterceptorChain.invoke(InvocationInterceptorChain.java:37)
	at org.junit.jupiter.engine.execution.InterceptingExecutableInvoker.invoke(InterceptingExecutableInvoker.java:92)
	at org.junit.jupiter.engine.execution.InterceptingExecutableInvoker.invoke(InterceptingExecutableInvoker.java:86)
	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.lambda$invokeTestMethod$7(TestMethodTestDescriptor.java:218)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.invokeTestMethod(TestMethodTestDescriptor.java:214)
	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.execute(TestMethodTestDescriptor.java:139)
	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.execute(TestMethodTestDescriptor.java:69)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$6(NodeTestTask.java:151)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:141)
	at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:137)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$9(NodeTestTask.java:139)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:138)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:95)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
	at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.invokeAll(SameThreadHierarchicalTestExecutorService.java:41)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$6(NodeTestTask.java:155)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:141)
	at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:137)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$9(NodeTestTask.java:139)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:138)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:95)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
	at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.invokeAll(SameThreadHierarchicalTestExecutorService.java:41)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$6(NodeTestTask.java:155)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:141)
	at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:137)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$9(NodeTestTask.java:139)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:138)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:95)
	at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.submit(SameThreadHierarchicalTestExecutorService.java:35)
	at org.junit.platform.engine.support.hierarchical.HierarchicalTestExecutor.execute(HierarchicalTestExecutor.java:57)
	at org.junit.platform.engine.support.hierarchical.HierarchicalTestEngine.execute(HierarchicalTestEngine.java:54)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:147)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:127)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:90)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.lambda$execute$0(EngineExecutionOrchestrator.java:55)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.withInterceptedStreams(EngineExecutionOrchestrator.java:102)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:54)
	at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:114)
	at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:86)
	at org.junit.platform.launcher.core.DefaultLauncherSession$DelegatingLauncher.execute(DefaultLauncherSession.java:86)
	at org.junit.platform.launcher.core.SessionPerRequestLauncher.execute(SessionPerRequestLauncher.java:53)
	at com.intellij.junit5.JUnit5IdeaTestRunner.startRunnerWithArgs(JUnit5IdeaTestRunner.java:57)
	at com.intellij.rt.junit.IdeaTestRunner$Repeater$1.execute(IdeaTestRunner.java:38)
	at com.intellij.rt.execution.junit.TestsRepeater.repeat(TestsRepeater.java:11)
	at com.intellij.rt.junit.IdeaTestRunner$Repeater.startRunnerWithArgs(IdeaTestRunner.java:35)
	at com.intellij.rt.junit.JUnitStarter.prepareStreamsAndStart(JUnitStarter.java:232)
	at com.intellij.rt.junit.JUnitStarter.main(JUnitStarter.java:55)
Caused by: com.puppycrawl.tools.checkstyle.api.CheckstyleException: Unable to resolve module name: LineLength. Please check for spelling errors or specify fully qualified class name.
	at com.puppycrawl.tools.checkstyle.bdd.InlineConfigParser.getFullyQualifiedClassName(InlineConfigParser.java:331)
	at com.puppycrawl.tools.checkstyle.bdd.InlineConfigParser.setModuleName(InlineConfigParser.java:391)
	at com.puppycrawl.tools.checkstyle.bdd.InlineConfigParser.handleXmlConfig(InlineConfigParser.java:277)
	at com.puppycrawl.tools.checkstyle.bdd.InlineConfigParser.setModules(InlineConfigParser.java:250)
	at com.puppycrawl.tools.checkstyle.bdd.InlineConfigParser.parse(InlineConfigParser.java:207)
	... 71 more


Process finished with exit code -1
```

Should we use `verify()` method instead of `verifyWithInlineConfigParser()`? The test case was working fine when we were using `verify()` method.

Comment (romani) [reply src/test/java/com/puppycrawl/tools/checkstyle/CheckerTest.java:442]: Please push your code to let me see all details

Comment (Zopsss) [reply src/test/java/com/puppycrawl/tools/checkstyle/CheckerTest.java]: sorry for the late reply, I have pushed the changes @romani

Comment (mahfouz72) [reply src/test/java/com/puppycrawl/tools/checkstyle/CheckerTest.java]: @Zopsss severity is a property of the Checker module not line Length I think that's why it says config comment isn't specified properly

Comment (Zopsss) [reply src/test/java/com/puppycrawl/tools/checkstyle/CheckerTest.java]: > @Zopsss severity is a property of the Checker module not line Length I think that's why it says config comment isn't specified properly

thats not true. Every module has this property, the implementation of this property is written at Checker check. Every modules are child of Checker so they can use this property to control audit events. Check out the [doc](https://checkstyle.sourceforge.io/config.html#Severity) for more info.

File: src/test/resources/com/puppycrawl/tools/checkstyle/checker/InputCheckerTestSeverity.java
Comment (romani) [inline src/test/resources/com/puppycrawl/tools/checkstyle/checker/InputCheckerTestSeverity.java:3]: to make it work add

```
        moduleMappings.put("LineLength",
                "com.puppycrawl.tools.checkstyle.checks.sizes.LineLengthCheck");
```

to InlineConfigParser.java

**Review by romani — CHANGES_REQUESTED** [review]
items:

**Review by romani — APPROVED** [review]
ok to merge if CI pass

File: src/test/resources/com/puppycrawl/tools/checkstyle/checker/InputCheckerTestSeverity.java
Comment (Zopsss) [reply src/test/resources/com/puppycrawl/tools/checkstyle/checker/InputCheckerTestSeverity.java:3]: added this as you suggested, now test case is passing but when I manually mutate the code and run it again it is still passing, this should not be happening, it should be failing... I have pushed the changes, can you inspect this pls?

the mutated code is as followed:

**before mutation:**
```java
public final void setSeverity(String severity) {
    this.severity = SeverityLevel.getInstance(severity);
}
```

**after mutation:**
```java
public final void setSeverity(String severity) {
}
```

**Comment by Zopsss** [comment]
> ok to merge if CI pass

no this is not okay, the test case should be failing after mutation please check this comment https://github.com/checkstyle/checkstyle/pull/14462#discussion_r1493357233

File: src/test/resources/com/puppycrawl/tools/checkstyle/checker/InputCheckerTestSeverity.java
Comment (Zopsss) [reply src/test/resources/com/puppycrawl/tools/checkstyle/checker/InputCheckerTestSeverity.java:3]: I have unresolved this conversation... waiting for your reply @romani

**Comment by romani** [comment]
> mutatedClass com.puppycrawl.tools.checkstyle.Checker

so you need to set property in Checker not in LineLength.
Play with:
```
<module name="Checker">
  <property name="severity" value="ignore"/>

```

**Comment by Zopsss** [comment]
> > mutatedClass com.puppycrawl.tools.checkstyle.Checker
> 
> so you need to set property in Checker not in LineLength. Play with:
> 
> ```
> <module name="Checker">
>   <property name="severity" value="ignore"/>
> ```

did this but it is still giving the same result, the test case is still passing even after mutation. Pushed the changes for you to review it.

**Comment by mahfouz72** [comment]
I don't why but when using xml config `<property name="severity" value="ignore"/>`
 pitest fails and the specified property of Checker (severity) Considered as not tested so the coverage drop for the Checker class and a mutation for this property in the checker class survived happened also at  https://github.com/checkstyle/checkstyle/pull/14492 and to me before

**Comment by Zopsss** [comment]
@romani ping

**Comment by Zopsss** [comment]
I tried debugging the code and found that `createConfiguration()` method used inside `verifyWithInlineConfigParser()` is not generating proper config. It is not adding the `severity` property to the Checker module. Then `verifyViolations()` method is using that inappropriate config to check the violation. We're expecting that there will be no violation as we have added the `severity` property and set it to `ignore` but the `createConfiguration()` is generating the config without `severity` property and it is expecting violation on line 11. That's why our test case is failing.

**Comment by romani** [comment]
lets try to move forward and fix code to make in-memory config to be correct (what we expect).

**Comment by Zopsss** [comment]
> lets try to move forward and fix code to make in-memory config to be correct (what we expect).

Can you explain me what do you mean by in-memory config? Do you mean the config created by `createConfiguration()`?

**Comment by Zopsss** [comment]
Also should I use verify() in this PR? Otherwise this PR will be open until the config generation is fixed. When we fix the config generation then we can convert this test and other tests to use the verifyWithInlineConfigParser()

@romani

**Comment by romani** [comment]
>Do you mean the config created by createConfiguration()?

Yes.

You can fix problem in this method in this PR, looks like it is first time we use config like this, do it will be natural to see it here and immediately use it in Input.

**Comment by Zopsss** [comment]
> >Do you mean the config created by createConfiguration()?
> 
> Yes.
> 
> You can fix problem in this method in this PR, looks like it is first time we use config like this, do it will be natural to see it here and immediately use it in Input.

Ohkay and should I keep it in seperate commit?

**Comment by romani** [comment]
in same commit is ok.

**Comment by Zopsss** [comment]
@romani I have pushed the changes, now our requirement is satisfied... We're able to add `severity` property to the generated config. The test case is also working fine, it is passing normally and failing when we mutate the code.

I had to make new `verifyWithInlineConfigParser()`, initially I tried using the default `verifyWithInlineConfigParser()` with some modification but then other input tests started failing so I had to make a new verify method.

File: src/test/resources/com/puppycrawl/tools/checkstyle/checker/InputCheckerTestSeverity.java
Comment (romani) [reply src/test/resources/com/puppycrawl/tools/checkstyle/checker/InputCheckerTestSeverity.java:4]: Uhh, can we reuse code that use xml config load from examples execution?
We do not need special code to parse , it is regular xml config, we already have code to load it properly.

Comment (Zopsss) [reply src/test/resources/com/puppycrawl/tools/checkstyle/checker/InputCheckerTestSeverity.java:4]: Do you mean by the new `createConfiguration()` method I created? It is there to check for the global properties and add it to the config. The old method was not checking for the global properties, I tried to modify the old method to check for the global properties but then other inputs examples tests started failing so I had to create the new method for that.

Comment (Zopsss) [reply src/test/resources/com/puppycrawl/tools/checkstyle/checker/InputCheckerTestSeverity.java:4]: Do you mean that we should use the old verifyWithInlineConfigParser() method & createConfiguration() method instead of the new one I made? I had to make them to add the global properties to the config, I've explained the reason here: https://github.com/checkstyle/checkstyle/pull/14462#issuecomment-1958094593 @romani

**Comment by Zopsss** [comment]
fixed a small bug in `createConfiguration()`, now our test case is passing and when we mutate the code it is failing as expected.

**Reason behind making new  `createConfiguration()`:

the regular createConfiguration() does not checks for the global properties, to check for the global properties we needed the whole xml config to see if it contains any global properties before any child modules, for this I created a new createConfiguration() which takes `filePath` as a parameter. With the help of `filePath` we can get the whole xml config, after getting it I ran a loop to check if it contains any global properties before starting a new child module, if it does then we add it to the `root` config, after checking the the xml config we call the regular createConfiguration() method which generates the config with global properties available to it.

File: src/test/resources/com/puppycrawl/tools/checkstyle/checker/InputCheckerTestSeverity.java
Comment (Zopsss) [reply src/test/resources/com/puppycrawl/tools/checkstyle/checker/InputCheckerTestSeverity.java:4]: do you mean by the new `createConfiguration()` method I created? if yes then here is the reason explaining that: https://github.com/checkstyle/checkstyle/pull/14462#issuecomment-1962315124 , I also created the `verifyWithInlineConfigParserGlobalProperites()` before but then I tried modifying the the current `verifyWithInlineConfigParser()` to use the new createConfiguration() and everything was working fine. @romani

**Comment by Zopsss** [comment]
@romani

**Comment by romani** [comment]
This is not good, we should reuse existing parser. I can not help on the go, I need big screen time to find solution and guide you

**Comment by Zopsss** [comment]
> This is not good, we should reuse existing parser. I can not help on the go, I need big screen time to find solution and guide you. It would be good if you find solution your self

We're using the normal `verifyWithInlineConfigParser()`, I've only created the new `createConfiguration()` it is using under the hood.  I tried to find the solution by modifying the current `createConfiguration()` method so that we don't need to create a new one but I failed. I will need to properly investigate into this, so until then can we use the normal `verify()` method instead of `verifyWithInlineCOnfigParser()`. Once I will find the solution I will let you know.

**Comment by Zopsss** [comment]
@romani can you tell me the reason why we should not use the newly created `createConfiguration()` method?🤔 Just had this doubt because the new method is solving our problem. I know we can find a solution without creating the new method but it might take some time, and my solution is easy to understand and solves our problem.

File: src/test/java/com/puppycrawl/tools/checkstyle/bdd/InlineConfigParser.java
Comment (Zopsss) [inline src/test/java/com/puppycrawl/tools/checkstyle/bdd/InlineConfigParser.java:250]: removed the `getChildren()` so that we can also add the properties of `Checker` module to the `testInputConfigBuilder` instead of directly adding the children module's properites.

File: src/test/java/com/puppycrawl/tools/checkstyle/bdd/InlineConfigParser.java
Comment (Zopsss) [inline src/test/java/com/puppycrawl/tools/checkstyle/bdd/InlineConfigParser.java:283]: if the current module is `Checker` then we're recursively calling the `handleXmlConfig()` method to set properties of it's children modules, this way the `testInputConfigBuilder` will contain Checker properties as well as Checker's children modules' properties.

**Comment by Zopsss** [comment]
@romani I tried different approach this time, I need your opinion on this. Is it okay or do we need to change something? The test case is passing and after mutating the code it is failing as we expected.

**Comment by Zopsss** [comment]
CircleCI is complaining, idk why though

File: src/test/java/com/puppycrawl/tools/checkstyle/bdd/InlineConfigParser.java
Comment (Zopsss) [inline src/test/java/com/puppycrawl/tools/checkstyle/bdd/InlineConfigParser.java:329]: Had to add Checker here, cuz some tests started throwing an error something like: 

`config comment is not defined properly `

**Comment by Zopsss** [comment]
@romani ping

resolved the merge conflicts, now I only need your opinion on this.

**Comment by Zopsss** [comment]
@romani

File: src/test/java/com/puppycrawl/tools/checkstyle/bdd/InlineConfigParser.java
Comment (romani) [reply src/test/java/com/puppycrawl/tools/checkstyle/bdd/InlineConfigParser.java:283]: it is weird that `if ("Checker".equals(moduleName)) {` is below of `if ("TreeWalker".equals(moduleName)) {`
but it is better to fix it at new PR, here is new issue on this - https://github.com/checkstyle/checkstyle/issues/14653

**Review by romani — APPROVED** [review]
I confirming that we can not do better solution for now, more refactoring is required.
https://github.com/checkstyle/checkstyle/issues/14653

but this will unblock all PRs that planning to use Checkers/Treewalker in their tests.

**Pull request merged**
